nx-X11/lib/include/xtrans/Xtranssock.c

```diff
@@ -2442,19 +2442,20 @@ TRANS(SocketUNIXConnect) (XtransConnInfo ciptr,
        (ciptr->peeraddr = malloc(namelen)) == NULL)
     {
         prmsg (1,
-	"SocketUNIXCreateListener: Can't allocate space for the addr\n");
+        "SocketUNIXCreateListener: Can't allocate space for the addr\n");
+        free(ciptr->peeraddr);
```

Comment:
malloc returned NULL, no need for a free() call

Reply:
done